data/crac/crac-impl/src/main/java/com/powsybl/openrao/data/cracimpl/GeneratorActionAdderImpl.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.openrao.data.cracimpl;
+
+import com.powsybl.action.Action;
+import com.powsybl.action.GeneratorActionBuilder;
+import com.powsybl.openrao.data.cracapi.networkaction.GeneratorActionAdder;
+
+import static com.powsybl.openrao.data.cracimpl.AdderUtils.assertAttributeNotNull;
+
+/**
+ * @author Baptiste Seguinot {@literal <baptiste.seguinot at rte-france.com>}
+ */
+public class GeneratorActionAdderImpl extends AbstractSingleNetworkElementActionAdderImpl<GeneratorActionAdder> implements GeneratorActionAdder {
+    private Double activePowerValue;
+
+    GeneratorActionAdderImpl(NetworkActionAdderImpl ownerAdder) {
+        super(ownerAdder);
+    }
+
+    @Override
+    public GeneratorActionAdder withActivePowerValue(double activePowerValue) {
+        this.activePowerValue = activePowerValue;
+        return this;
+    }
+
+    protected Action buildAction() {
+        return new GeneratorActionBuilder()
+            .withId(String.format("%s_%s_%s", getActionName(), networkElementId, activePowerValue))
+            .withNetworkElementId(networkElementId)
+            .withActivePowerRelativeValue(false)
```

Comment:
No possibility / need to have a relative value change action ?

Reply:
For this Issue, it was only ask to keep the same functionality than before, no more.
If we want more, it will need a lot of changes: updates adder, tests, json... it needs a separate PR since this one is already huge. NB. There is a lot more possibilities for a GeneratorAction than what Rao used here.

Reply:
Relative value changes are not currently supported in the RAO because it adds quite a lot of complexity (currently we can apply the same action multiple times and it will have the same effect as applying it once, which would not be the case with a relative change)